app/Export/GetImages.hs

```diff
@@ -28,101 +28,94 @@ getActiveGraphImage req = do
-getScheduleByTime :: [CourseInfo] -> [CourseInfo] -> [[String]]
-getScheduleByTime lecTimes tutTimes =
-  let allTimes = lecTimes ++ tutTimes
+-- | Creates a schedule.
+-- It takes information about lectures and tutorials and their corresponding time.
+-- Courses are added to schedule, based on their days and times.
+getScheduleByTime :: [(String, String, String)] -> [(String, String, String)] -> [[Time]] -> [[Time]] -> [[String]]
+getScheduleByTime selectedLecs selectedTuts lecTimes tutTimes =
+  let lecture_times = zip selectedLecs lecTimes
+      tutorial_times = zip selectedTuts tutTimes
+      allTimes = lecture_times ++ tutorial_times
       schedule = replicate 13 $ replicate 5 ""
-      schedule' = foldl addCourseToSchedule schedule allTimes
-  in schedule'
-
--- CourseInfo {code = "STA355H1", section = "L0101", session = "F", time = [Time {timeField = [0.0,14.0]},Time {timeField = [0.0,14.5]},Time {timeField = [0.0,15.0]}]}
-addCourseToSchedule :: [[String]] -> CourseInfo -> [[String]]
-addCourseToSchedule schedule courseInfo
-    | (time courseInfo) == [] = schedule
-    | otherwise =
-        let time' = filter (\t-> (mod' (timeField t !! 1) 1) == 0) (time courseInfo)
-            timeArray = convertTimeToArray time'
-            in foldl (addCourseHelper (code courseInfo) (section courseInfo) (session courseInfo)) schedule timeArray
-
--- | Take a list of Time and returns a list of tuples that correctly index into the 2-D table (for generating the image)
--- Time {timeField = [0.0,14.0]} -> (0, 6)
+  in foldl addCourseToSchedule schedule allTimes
+
+-- | Take a list of Time and returns a list of tuples that correctly index
+-- into the 2-D table (for generating the image)
 convertTimeToArray :: [Time] -> [(Int, Int)]
 convertTimeToArray = map (\x -> (floor $ timeField x !! 0 , floor $ timeField x !! 1 - 8))
 
+addCourseToSchedule :: [[String]] -> ((String, String, String), [Time]) -> [[String]]
+addCourseToSchedule schedule (course, courseTimes) =
+  let time' = filter (\t-> (mod' (timeField t !! 1) 1) == 0) courseTimes
+      timeArray = convertTimeToArray time'
+  in foldl (addCourseHelper course) schedule timeArray
 
---  "STA355H1" "L0101" "F"  Time {timeField = [0.0,14.0]}
-addCourseHelper :: String -> String -> String -> [[String]] -> (Int, Int) -> [[String]]
-addCourseHelper courseCode courseSection courseSession currentSchedule (day, courseTime) =
+-- | Appends information of course to the current schedule for specified day and time.
+-- Returns new schedule.
+addCourseHelper :: (String, String, String) -> [[String]] -> (Int, Int) -> [[String]]
+addCourseHelper (courseCode, courseSection, courseSession) currentSchedule (day, courseTime) =
   let time_schedule = currentSchedule !! courseTime
       current_schedule = if (null $ time_schedule !! day) then (courseCode++courseSession++" "++courseSection) else (time_schedule !! day ++ ("&"++courseCode++courseSession++" "++courseSection))
```

Comment:
I believe we talked about storing the contents of each cell as a list of courses, rather than a formatted string.